src/Core/Content/Cms/Subscriber/CmsVersionMergeSubscriber.php

```diff
@@ -0,0 +1,80 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Cms\Subscriber;
+
+use Shopware\Core\Framework\DataAbstractionLayer\Event\BeforeVersionMergeEvent;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @internal
+ */
+#[Package('buyers-experience')]
+class CmsVersionMergeSubscriber implements EventSubscriberInterface
```

Comment:
Overall we are a bit concerned that this is rather a general problem and not one specific to CMS entities?

Do you think the same issue could occur with other entities?

If so, do you think the solution could be generalised? Maybe by inspecting the entity definitions and relationships?

If the answer is yes but you have other pressures and constraints we would be okay merging this but we would like to add some documentation/gotchas to the code in case similar issues arise in the future:

1. We mark the event internal, so that we don't have to support it forever when/if we find a general solution
2. We document the event dispatch & listener with information explaining the above situation. We would describe that it's a kind of hack and if a similar issue occurs in the future, we should take a step back and see if we can solve it on a general level.

What do you think?

Reply:
At this stage,  i believe this issue is specific to CMS , and even within the CMS , it manifests as a corner case . for example , this occurs only when certain steps are followed during versioning and merge process. I would prefer to wait until we identify concrete patterns or similar issues in other entities before generalize it.I'll add some documentation to the event & listener so we know that it's a specific case for the CMS

Reply:
Great, fine for us then :)